src/python-common/ceph/deployment/service_spec.py

```diff
@@ -173,11 +173,23 @@ def set_hosts(self, hosts):
         # in the orchestrator backend.
         self.hosts = hosts
 
-    def pattern_matches_hosts(self, all_hosts):
-        # type: (List[str]) -> List[str]
-        if not self.host_pattern:
+    def filter_matching_hosts(self, _get_hosts_func: Callable) -> List[str]:
```

Comment:
What about simplifying this to
```suggestion
    def filter_matching_hosts(self, hosts: List[HostSpec]) -> List[str]:
```
? Is there a need to encapsulate this in a function?

Reply:
Yes, this reduces the amount of calls to `get_hosts` as we can pass the arguments to it _inside_ `filter_matching_hosts()`. If we pass a hostlist to the function, we have to construct it before based on the presence of `labels`, `host_pattern` etc..